server/src/main/java/org/elasticsearch/transport/RemoteClusterAware.java

```diff
@@ -77,25 +97,37 @@ protected RemoteClusterAware(Settings settings) {
         this.clusterNameResolver = new ClusterNameExpressionResolver(settings);
     }
 
-    protected static Map<String, List<Supplier<DiscoveryNode>>> buildRemoteClustersSeeds(Settings settings) {
+    protected static Map<String, Tuple<String, List<Supplier<DiscoveryNode>>>> buildRemoteClustersSeeds(Settings settings) {
```

Comment:
Maybe a doc comment here? With the new work, it is really tough to figure out what is going on in this method.